src/server/common_services/auth_server.js

```diff
@@ -519,13 +511,20 @@ function _prepare_auth_request(req) {
         // for anonymous access operations
         const allow_anonymous_access = (options.anonymous === true);
 
+        dbg.log1('load_auth:', options, req.auth);
+        if (req.auth) {
+            req.account = system_store.data.get_by_id(req.auth.account_id);
+            req.system = system_store.data.get_by_id(req.auth.system_id);
+            req.role = req.auth.role;
+        }
+
         if (allow_anonymous_access) {
-            req.anonymous = true;
             // Currently authorize anonymous with the system that we have
-            if (!ignore_missing_system) {
+            if (!ignore_missing_system && !req.system) {
```

Comment:
Move below all other checks for account and system, and merge conditions to one - and explain in a single commet "Initialize default system for anynomous requests..."